pinot-common/src/main/java/org/apache/pinot/common/assignment/InstancePartitions.java

```diff
@@ -58,25 +60,37 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class InstancePartitions {
   private static final char PARTITION_REPLICA_GROUP_SEPARATOR = '_';
+  private static final String PARTITIONS_KEY = "partitions";
+  private static final String INSTANCE_SEPARATOR = "/";
 
   private final String _instancePartitionsName;
-  private final Map<String, List<String>> _partitionToInstancesMap;
+  private final Map<String, List<String>> _partitionWithReplicaGroupToInstancesMap;
+  private final Map<Integer, List<String>> _partitionToInstancesMap;
```

Comment:
Is the key in `_partitionToInstancesMap` the partitionID ?

Reply:
I did think about reusing the existing InstancePartitions from ZNode, but the problem is that there is no way to know about the pool information of an instance if its instance config is gone from the cluster; we don't know which pool the gone instance belongs to, so we don't know which pool we should pick the new instance from.
On the other hand, since we can choose which table should we enable this feature, the overhead on the ZNode should be controllable.